### ISSUE TITLE
DataCloneError: The object cannot be cloned in Safari v17 (using fetch-based MSW)

### ISSUE THREAD

**Issue by thepassle** [kettanaito/msw#1757]
### Prerequisites

- [X] I confirm my issue is not in the [opened issues](https://github.com/mswjs/msw/issues)
- [X] I confirm the [Frequently Asked Questions](https://mswjs.io/docs/faq) didn't contain the answer to my issue

### Environment check

- [X] I'm using the [latest](https://github.com/mswjs/msw/releases/latest) `msw` version
- [X] I'm using Node.js version 14 or higher

### Browsers

Safari

### Reproduction repository

https://github.com/thepassle/msw-safari-repro

### Reproduction steps

- `npm i`
- `npm start`
- open browser in safari 17

### Current behavior

<img width="1277" alt="repro" src="https://github.com/mswjs/msw/assets/17054057/1f042b1b-80d9-45be-a8fd-1787d6bd4e0a">

### Expected behavior

No error and the object is cloned successfully

**Comment by thepassle** [comment]
Managed to hunt it down to this code:

<img width="1267" alt="Screenshot 2023-10-03 at 11 20 26" src="https://github.com/mswjs/msw/assets/17054057/8d6688a1-b719-4f64-a81c-0ab9ac45808f">

It seems like its not possible anymore in Safari 17 to postmessage a `ReadableStream`. both `transfer` and `data.body` are `ReadableStream`s. When I set those values to something else it does work. So it looks like `postMessage` in combination with `ReadableStream` is broken.

Im not really sure this is a MSW bug, it kind of seems like a safari bug?
